### PR TITLE
Enhancements in React Vanilla Components

### PR DESCRIPTION
**Pull request by pavi41**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/react-vanilla-components/__tests__/components/CheckBoxGroup.test.tsx [hunk 1]
```diff
@@ -18,7 +18,16 @@ const field = {
   },
   fieldType: 'checkbox-group',
   enum: [1, 2, 3],
-  enumNames: ['checkbox 1', 'checkbox 2', 'checkbox 3']
+  enumNames: [ {
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/__tests__/components/CheckBoxGroup.test.tsx:21]: please add test cases for richtext

File: packages/react-vanilla-components/__tests__/components/DropDown.test.tsx [hunk 2]
```diff
@@ -19,7 +19,15 @@ const field = {
   fieldType: "drop-down",
   placeholder: "select",
   enum: [1, 2, 3],
-  enumNames: ["option 1", "option 2", "option 3"],
+  enumNames: [{
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/__tests__/components/DropDown.test.tsx:22]: please add test cases for richtext

File: packages/react-vanilla-components/__tests__/components/RadioButtonGroup.test.tsx [hunk 3]
```diff
@@ -18,7 +19,16 @@ const field = {
   },
   fieldType: 'radio-group',
   enum: [1, 2, 3],
-  enumNames: ['radio 1', 'radio 2', 'radio 3']
+  enumNames: [{
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/__tests__/components/RadioButtonGroup.test.tsx:22]: please add test cases for richtext

File: packages/react-vanilla-components/src/components/CheckBoxGroup.tsx [hunk 4]
```diff
@@ -73,8 +74,8 @@ const CheckBoxGroup = (props: PROPS) => {
           className={`cmp-adaptiveform-checkboxgroup__widget ${orientation}`}
           id={`${id}-widget`}
         >
-          {options?.map((item: string, index: number) => (
-            <div className={`cmp-adaptiveform-checkboxgroup-item ${name}`} key={item}>
+          {options?.map((item, index: number) => (
+            <div className={`cmp-adaptiveform-checkboxgroup-item ${name}`} key={`${item.value}-${index}`}>
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/src/components/CheckBoxGroup.tsx:78]: Don’t use index in key

File: packages/react-vanilla-components/src/components/DropDown.tsx [hunk 5]
```diff
@@ -55,12 +56,14 @@ const DropDown = (props: PROPS) => {
...
           disabled={!enabled}
+          defaultValue={'DEFAULT'}
         >
-          <option value="" disabled selected>{placeholder}</option>
+          <option value="DEFAULT" disabled>{placeholder}</option>
           {
-            dropValue?.map((item: string, index: number) => {
-              return <option className="cmp-adaptiveform-dropdown__option" key={item} value={enums![index]}>{item}</option>;
+            dropValue?.map((item, index: number) => {
+              return <option className="cmp-adaptiveform-dropdown__option" key={`${item.value}-${index}`} value={enums![index]}>{item?.value}</option>;
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/src/components/DropDown.tsx:66]: Don’t use index in key

File: packages/react-vanilla-components/src/components/common/Label.tsx [hunk 6]
```diff
@@ -34,7 +34,7 @@ const Label = (props: LABEL) => {
           <button
             className={`${bemBlock}__questionmark`}
             onClick={onClick}
-            aria-label={description}
+            title="Help Text"
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/src/components/common/Label.tsx:37]: please "aria-label" and pass description in title

File: packages/react-vanilla-components/src/components/TelephoneInput.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import React, {useCallback} from 'react';
+import { PROPS } from '../utils/type';
+import FieldWrapper from './common/FieldWrapper';
+import { withRuleEngine } from '../utils/withRuleEngine';
+
+const TelephoneInput = (props: PROPS) => {
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/src/components/TelephoneInput.tsx:6]: please add dispatchBlur & dispatchFocus event

File: packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    pattern: "^[+]1[0-9]{0,10}$",
+    label: {
+      value: "Telephone Input",
+      visible: true
+    },
+    ":type": "forms-components-examples/components/form/telephoneinput"
+};
+const helper = renderComponent(TelephoneInput);
+
+describe('Text Field', () => {
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx:36]: please rename test cases name

File: packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      visible: false,
+    };
+    const { renderResponse } = await helper(f);
+    expect(renderResponse.queryByText(f.label.value)).toBeNull();
+  });
+
+  test('error message element exists when the field is invalid', async () => {
+    const f = {
+      ...field,
+      valid: false,
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx:62]: Avoid passing hard-coded values for the "valid" property.

File: packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL ADOBE NOR ITS THIRD PARTY PROVIDERS AND PARTNERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import TelephoneInput from "../../src/components/TelephoneInput";
+import userEvent from "@testing-library/user-event";
+import { renderComponent, DEFAULT_ERROR_MESSAGE } from '../utils';
+import "@testing-library/jest-dom/extend-expect"
+
+const field = {
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx:14]: please add test cases for invalid test cases

File: packages/react-vanilla-components/src/utils/withRuleEngine.tsx [hunk 11]
```diff
@@ -50,6 +50,17 @@ const getLocalizePlaceholder = (i18n: any, state: FieldViewState) => {
   return placeholder;
 };
 
+const getLocalizeEnumNames = (i18n: any, state: FieldViewState) => {
```
Comment (amit-mnnit) [inline packages/react-vanilla-components/src/utils/withRuleEngine.tsx:53]: Please verify "getLocalizeEnumNames" method for localization

**Review by amit-mnnit — CHANGES_REQUESTED** [review]
please check comments

File: packages/react-vanilla-components/src/utils/withRuleEngine.tsx (hunk 11 above)
Comment (pavi41) [reply packages/react-vanilla-components/src/utils/withRuleEngine.tsx:53]: Done!

File: packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx (hunk 10 above)
Comment (pavi41) [reply packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx:14]: Added!

File: packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx (hunk 9 above)
Comment (pavi41) [reply packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx:62]: Done!

File: packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx (hunk 8 above)
Comment (pavi41) [reply packages/react-vanilla-components/__tests__/components/TelephoneInput.test.tsx:36]: Done!

File: packages/react-vanilla-components/src/components/TelephoneInput.tsx (hunk 7 above)
Comment (pavi41) [reply packages/react-vanilla-components/src/components/TelephoneInput.tsx:6]: Added!

File: packages/react-vanilla-components/src/components/common/Label.tsx (hunk 6 above)
Comment (pavi41) [reply packages/react-vanilla-components/src/components/common/Label.tsx:37]: Added!

File: packages/react-vanilla-components/src/components/DropDown.tsx (hunk 5 above)
Comment (pavi41) [reply packages/react-vanilla-components/src/components/DropDown.tsx:66]: Changed!

File: packages/react-vanilla-components/src/components/CheckBoxGroup.tsx (hunk 4 above)
Comment (pavi41) [reply packages/react-vanilla-components/src/components/CheckBoxGroup.tsx:78]: Changed!

File: packages/react-vanilla-components/__tests__/components/RadioButtonGroup.test.tsx (hunk 3 above)
Comment (pavi41) [reply packages/react-vanilla-components/__tests__/components/RadioButtonGroup.test.tsx:22]: Added!

File: packages/react-vanilla-components/__tests__/components/CheckBoxGroup.test.tsx (hunk 1 above)
Comment (pavi41) [reply packages/react-vanilla-components/__tests__/components/CheckBoxGroup.test.tsx:21]: Added!

File: packages/react-vanilla-components/__tests__/components/DropDown.test.tsx (hunk 2 above)
Comment (pavi41) [reply packages/react-vanilla-components/__tests__/components/DropDown.test.tsx:22]: There is no richtext for options in dropdown!

**Review by amit-mnnit — APPROVED** [review]

**Pull request merged**